pylint/checkers/classes/class_checker.py

```diff
@@ -467,8 +467,7 @@ def _is_attribute_property(name: str, klass: nodes.ClassDef) -> bool:
         if inferred.pytype() != property_name:
             continue
 
-        cls = node_frame_class(inferred)
-        if cls == klass.declared_metaclass():
+        if (_cls := node_frame_class(inferred)) == klass.declared_metaclass():
```

Comment:
Keeping it separate is better here IMO

Reply:
Why? That variable isn't used anywhere else.

Reply:
There's seven concepts on the line (variable name, assignment expression, function call, function args, equal operator, namespace, function call) and a human can handle 5 to 7 at once on a good day. It's too much info packed on the same line.

Reply:
Okay, I cut out the unused variable. Now the line looks like this:

```python
        if node_frame_class(inferred) == klass.declared_metaclass():
            continue
```
It is now logically simpler than a clause just above it:
```python
        if isinstance(inferred, nodes.FunctionDef) and decorated_with_property(
            inferred
        ):
            return True
```
It seems like there's a general perception that all this code is already written in an optimally readable style. But in my experience that's far from the case.

Reply:
> It seems like there's a general perception that all this code is already written in an optimally readable style. 

There is above all a very high risk of introducing bug with this kind of stylistic changes. The code as it is might not be "optimally readable" but it is **good enough** and has the benefit of being battle tested. 

> If there truly is a desire for refactoring, I would suggest to break it up into smaller PRs to be able to review it better. Furthermore, I would suggest to limit the changes to the most obvious examples where a clear benefit is visible.

Can't agree more with that statement.

> I think it would make sense to disable the consider-using-assignment-expression check be default. As explained earlier, it just can't detect these nuances that are required here.

We can also learn a lot about what is wrong with the check atm, in particular I saw a false positive on a list comprehension that we had to disable, there are line that would be too complex or too long if we apply the suggestion blindly. This is an opportunity to make this check better.

Reply:
> > I think it would make sense to disable the consider-using-assignment-expression check be default. As explained earlier, it just can't detect these nuances that are required here.
> 
> We can also learn a lot about what is wrong with the check atm, in particular I saw a false positive on a list comprehension that we had to disable, there are line that would be too complex or too long if we apply the suggestion blindly. This is an opportunity to make this check better.

I did attempt to do this at one point but it never felt good enough, so I didn't create a PR for it. One of the keys seems to be to make it more customizable but this undoubtedly adds (a lot) of complexity. This is my working branch: https://github.com/pylint-dev/pylint/compare/main...cdce8p:customize_assignment-expr

Just for reference the config I tested for it with Home Assistant.
<details>
<summary>pylint config</summary>

```
[tool.pylint.CODE_STYLE]
max-line-length-suggestions = 72
assignment-expr-not = true
assignment-expr-compare-none = true
assignment-expr-compare-const = true
assignment-expr-compare-in = false
assignment-expr-compare-call = false
assignment-expr-assign-not = true
assignment-expr-assign-getitem = true
assignment-expr-assign-annotation = false
assignment-expr-assign-await = true
assignment-expr-assign-ifexp = false
assignment-expr-assign-comprehension = false
assignment-expr-assign-lambda = false
assignment-expr-assign-compare = false
assignment-expr-assign-boolop = false
assignment-expr-assign-binop = false
assignment-expr-assign-call = true
assignment-expr-assign-call-func-names = [
#    "get",
#    "async_get",
#    "async_get_last_state",
    "search",
    "match",
    "parse_datetime",
    "parse_time",
    "parse_date",
    "async_load",
#    "get_entity",
    "async_prepare_reload",
    "pop",
    "getattr",
]
assignment-expr-assign-call-func-names-rgxs = [
    "^_?(async_)?get(_.*)?$",
    "^_?convert(_.*)?$",
]
```
</details>